low-level/framework/utils/consulstore.py

```diff
@@ -93,48 +141,76 @@ def get(self, key, **kwargs):
             except requests.exceptions.ConnectionError as connerr:
                 logger.warn("Error[{0}] consul connection refused Retry Index {1}" \
                     .format(connerr, retry_index))
+                self._consul_conn_status = False
                 time.sleep(WAIT_BEFORE_RETRY)
 
             except Exception as gerr:
                 logger.warn("Error[{0}] while reading data from consul {1}" \
-                    .format(gerr, key))
+                    .format(gerr, consul_key))
+                self._consul_conn_status = False
                 break
+
+        return data
+
+    def get(self, key, **kwargs):
+        """Load data from given key."""
+        data = None
+        if self._consul_conn_status is False:
+            self._establish_consul_connection(self._host, self._port)
+
+        if self._consul_conn_status and self._data_sync_required:
+            self._dump_filestore_to_consulstore()
+
+        data = self._consul_store_get(key, kwargs)
+
+        if self._consul_conn_status is False:
+            data = self._file_store.get(key,kwargs)
+
         return data
 
     def exists(self, key):
-        """check if key exists
-        """
+        """Check if key exists."""
         if self.get(key):
             return True
         else:
-            return False
+            return self._file_store.exists(key)
 
-    def delete(self, key):
-        """ delete a key
-        """
+    def _consul_store_delete(self, key):
+        """Delete a key from consul store."""
         for retry_index in range(0, MAX_CONSUL_RETRY):
             try:
-                key = self._get_key(key)
-                self.consul_conn.kv.delete(key)
+                consul_key = self._get_key(key)
+                self.consul_conn.kv.delete(consul_key)
+                self._consul_conn_status = True
                 break
 
             except requests.exceptions.ConnectionError as connerr:
                 logger.warn("Error[{0}] consul connection refused Retry Index {1}" \
                     .format(connerr, retry_index))
+                self._consul_conn_status = False
                 time.sleep(WAIT_BEFORE_RETRY)
 
             except Exception as gerr:
                 logger.warn("Error[{0}] while deleting key from consul {1}" \
-                    .format(gerr, key))
+                    .format(gerr, consul_key))
+                self._consul_conn_status = False
                 break
 
-    def get_keys_with_prefix(self, prefix):
-        """ get keys with given prefix
-        """
+    def delete(self, key):
+        """Delete a key."""
+        self._consul_store_delete(key)
+        self._file_store.delete(key)
+        if self._consul_conn_status is False:
```

Comment:
this should happen in _consul_store_delete() itself

Reply:
done